Compile failure on Ubuntu Trusty (14.04) with AVFORMAT

With `--enable-avformat`, libyami-utils fails to compile using the Ubuntu Trusty provided libav*-dev packages.  It compiles fine on Ubuntu Xenial (16.04) and Yakkety (16.10).  Configure should detect the proper version of libav* required to compile.

> ../tests/decodeinputavformat.cpp:106:5: error: 'AV_CODEC_ID_H265' was not declared in this scope
> AV_CODEC_ID_H265, YAMI_MIME_H265,
> ^
> ../tests/decodeinputavformat.cpp: In member function 'virtual bool DecodeInputAvFormat::getNextDecodeUnit(VideoDecodeBuffer&)':
> ../tests/decodeinputavformat.cpp:127:34: error: 'av_packet_unref' was not declared in this scope
> av_packet_unref(&m_packet);
> ^
> ../tests/decodeinputavformat.cpp: In destructor 'virtual DecodeInputAvFormat::~DecodeInputAvFormat()':
> ../tests/decodeinputavformat.cpp:153:34: error: 'av_packet_unref' was not declared in this scope
> av_packet_unref(&m_packet);
> ^
> make[2]: *** [../tests/grid-decodeinputavformat.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> ../tests/decodeinputavformat.cpp:106:5: error: 'AV_CODEC_ID_H265' was not declared in this scope
> AV_CODEC_ID_H265, YAMI_MIME_H265,
> ^
> ../tests/decodeinputavformat.cpp: In member function 'virtual bool DecodeInputAvFormat::getNextDecodeUnit(VideoDecodeBuffer&)':
> ../tests/decodeinputavformat.cpp:127:34: error: 'av_packet_unref' was not declared in this scope
> av_packet_unref(&m_packet);
> ^
> ../tests/decodeinputavformat.cpp: In destructor 'virtual DecodeInputAvFormat::~DecodeInputAvFormat()':
> ../tests/decodeinputavformat.cpp:153:34: error: 'av_packet_unref' was not declared in this scope
> av_packet_unref(&m_packet);
> ^
> make[2]: *** [../tests/decodeinputavformat.o] Error 1

--- a/tests/decodeinputavformat.cpp
+++ b/tests/decodeinputavformat.cpp
@@ -22,6 +22,10 @@
 #include "common/log.h"
 #include <Yami.h>
 
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(55, 39, 100)
+#define av_packet_unref av_free_packet
+#endif
+
 DecodeInputAvFormat::DecodeInputAvFormat()
 :m_format(NULL),m_videoId(-1), m_codecId(AV_CODEC_ID_NONE), m_isEos(true)
 {
@@ -103,7 +107,9 @@ static const MimeEntry MimeEntrys[] = {
 #endif
 
     AV_CODEC_ID_H264, YAMI_MIME_H264,
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55, 39, 100)
     AV_CODEC_ID_H265, YAMI_MIME_H265,
+#endif
     AV_CODEC_ID_WMV3, YAMI_MIME_VC1,
     AV_CODEC_ID_VC1, YAMI_MIME_VC1
 };

